app/controllers/APIController.scala

```diff
@@ -67,6 +67,35 @@ class APIController @Inject() (
       }
   }
 
+  def updateProduct(id: String) = Action.async {
+    implicit request: Request[AnyContent] =>
+      bakeryDB.getProductById(id).map {
+        case None => NotFound("No Product Found")
+        case Some(myProduct) =>
+          val requestJson = request.body.asJson
+          Future {
+            requestJson match {
+              case None => BadRequest("No request body found")
+              case Some(updateProduct) => {
+                val ProductJsonResult =
+                  updateProduct.validate[productUpdateRequest]
+                if (ProductJsonResult.isSuccess) {
+                  val rowsUpdated =
+                    bakeryDB.updateProduct(myProduct.id, updateProduct)
```

Comment:
If you know `ProductJsonResult.isSuccess` is true, then you can do `ProductJsonResult.get` (it might be `ProductJsonResult.value` ) and just pass the request object to the function itself.